ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/RequestHandler.java

```diff
@@ -0,0 +1,141 @@
+  private final KsqlEngine ksqlEngine;
+  private final KsqlConfig ksqlConfig;
+  private final ServiceContext serviceContext;
+  private final DistributingExecutor distributor;
+  private final CommandQueue commandQueue;
+  private final Duration distributedCommandSyncTimeout;
+
+  /**
+   * @param customExecutors a map describing how to execute statements that do not need
+   *                        to be distributed onto the command topic
+   * @param mustSynchronize a predicate describing which statements must wait for previous
+   *                        distributed statements to finish before handling (a value of
+   *                        {@code true} will require synchronization)
+   * @param distributor     the distributor to use if there is no custom executor for a given
+   *                        statement
+   * @param ksqlEngine      the primary KSQL engine - the state of this engine <b>will</b>
+   *                        be directly modified by this class
+   * @param ksqlConfig      a configuration
+   * @param serviceContext  a service context
+   * @param commandQueue    the command queue to distribute to
+   * @param distSyncTimeout the maximum amount of time to wait for previously distributed
+   */
+  public RequestHandler(
+      final Map<Class<? extends Statement>, StatementExecutor> customExecutors,
+      final Predicate<Class<? extends Statement>> mustSynchronize,
+      final DistributingExecutor distributor,
+      final KsqlEngine ksqlEngine,
+      final KsqlConfig ksqlConfig,
+      final ServiceContext serviceContext,
+      final CommandQueue commandQueue,
+      final Duration distSyncTimeout
+  ) {
+    this.customExecutors = customExecutors;
+    this.mustSynchronize = mustSynchronize;
+    this.ksqlEngine = ksqlEngine;
+    this.ksqlConfig = ksqlConfig;
+    this.serviceContext = serviceContext;
+    this.distributor = distributor;
+    this.commandQueue = commandQueue;
+    this.distributedCommandSyncTimeout = distSyncTimeout;
+  }
+
+  public KsqlEntityList execute(
+      final List<ParsedStatement> statements,
+      final Map<String, Object> propertyOverrides
+  ) {
+    final Map<String, Object> scopedPropertyOverrides = new HashMap<>(propertyOverrides);
+    final KsqlEntityList entities = new KsqlEntityList();
+    for (ParsedStatement statement : statements) {
+      final KsqlEntity result = executeStatement(statement, scopedPropertyOverrides, entities);
+      if (result != null) {
+        entities.add(result);
+      }
+    }
+    return entities;
+  }
+
+  private KsqlEntity executeStatement(
+      final ParsedStatement parsed,
+      final Map<String, Object> propertyOverrides,
+      final KsqlEntityList entities
+  ) {
+    final PreparedStatement prepared = ksqlEngine.prepare(parsed);
+    final Class<? extends Statement> statementClass = prepared.getStatement().getClass();
+
+    if (mustSynchronize.test(statementClass)) {
```

Comment:
I'd move the contents of this if statement into a function to make this function more readable.